### PR TITLE
I-Day add unit tests

### PR DESCRIPTION
**Pull request by jflinchbaugh**
Introduce unit testing for helm charts.

build: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/Helm3%20Charts/job/Feature%20Snapshot%20Builds/job/i-day-unit-test/

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# and you may not use this file except in compliance with the Apache License Version 2.0.
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+#
+
+FROM docker-all.repo.sonatype.com/alpine/helm:3.7.2
```
Comment (jflinchbaugh) [inline Dockerfile:14]: new version of helm

**Review by eduard-tita — APPROVED** [review]
I have to read more about it, but it looks like a really good start.

File: charts/nexus-iq/tests/deployment_test.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+suite: deployment
```
Comment (hectorhuol) [inline charts/nexus-iq/tests/deployment_test.yaml:1]: @jflinchbaugh this is awesome! Just qq, it is possible to check the contents of the logs? I am thinking that maybe in the future we could dynamically check the IQ version from the logs, which will help us to fully automate the release process.

**Review by hectorhuol — APPROVED** [review]
+1

File: charts/nexus-iq/tests/deployment_test.yaml (hunk 2 above)
Comment (jflinchbaugh) [reply charts/nexus-iq/tests/deployment_test.yaml:1]: there _is_ a capability built-in for running integration tests, but it requires that the helm chart be installed into a cluster, and then you can run the tests as pods/jobs alongside the running components. that might be the place to do such a thing, though, it requires a running cluster. maybe we can figure out a minikube instance in jenkins or use sonatype's cloudy environment at some point.

**Pull request merged**